packages/shr-fhir-export/lib/extensions.js

```diff
@@ -538,6 +552,51 @@ class ExtensionExporter {
     }
   }
 
+  inlineSubextension(subExt, extension, field, baseId, baseExt) {
+    for (const ext of subExt.snapshot.element) {
+      const snapExt = common.cloneJSON(ext);
+      // Slice off the extraneous "Extension" that will start the id
+      const splitPoint = snapExt.id.indexOf('.');
+      const endId = snapExt.id.slice(splitPoint + 1);
+      // Construct a newId and path for the inlined version of the extension
+      const newId = `${baseId}.extension:${common.shortID(field.effectiveIdentifier)}.${endId}`;
+      const pathEnd = endId.slice(endId.lastIndexOf('.') + 1);
+      const existingElement = extension.snapshot.element.find(el => el.id === newId);
+      // Applying constraints can add elements that we don't actually want, just keep values
+      if (!pathEnd.startsWith('value')) {
+        extension.snapshot.element = extension.snapshot.element.filter(el => el.id !== newId);
+      }
+      // This avoids inlining the uneccessary top level "Extension" element or readding value that was constrained
+      if (splitPoint < 0 || (existingElement != null && pathEnd.startsWith('value'))) {
+        continue;
+      } else {
+        // Set the inlined version of the snapshot element with the path and id
+        snapExt.id = newId;
+        snapExt.path = this.getExtensionPathFromExtensionID(snapExt.id);
+        if (pathEnd === 'url') {
+          // Fix the "url" field to the name of the inlined extension, following example of USCore
+          snapExt.fixedUri = newId.slice(newId.lastIndexOf(':') + 1, newId.lastIndexOf('.'));
+        }
+      }
+      // Find the correct spot for the snapshot element and splice it in 
+      let i = extension.snapshot.element.findIndex(e => e == baseExt) + 1;
+      for ( ; i < extension.snapshot.element.length && extension.snapshot.element[i].path.startsWith(baseExt.path); i++);
+      extension.snapshot.element.splice(i, 0, snapExt);
```

Comment:
I found it difficult to understand the for-loop at first. Could this be revised to avoid using the for-loop in order to get `i` to the desired splice index? Please let me know if you want to talk about how to do this.